odoo/osv/query.py

```diff
@@ -130,6 +130,22 @@ def select(self, *args):
         )
         return query_str, params
 
+    def subselect(self, *args):
+        """ Similar to :meth:`select`, but for sub-queries.
+            This one avoids the ORDER BY clause when possible.
+        """
+        if self.limit or self.offset:
```

Comment:
Maybe `order` as well so it's possible to explicitly opt into ordering on subselect by... ordering them?

Reply:
That wouldn't work, actually.  The subselects actually come from calls to `_search()` where ordering is mandatory: searching with `order=None` or `order=False`, you fall back on the model's default ordering, therefore all those calls to `_search` return a `Query` with an `order` :frowning_face:  Of course, we could follow each call to `_search` in expression.py by a `result.order = None`, but that would be a mess.

The _right thing_ would be to explicitly ask for no ordering when calling `_search`, but that's a nontrivial behavior change.  We need a much clearer way to distinguish between _no ordering_ and _the default ordering_.

Reply:
:(